### PR TITLE
Import and run `ansible-hardening`

### PR DESCRIPTION
**Pull request by NicolasT**
This commit imports the OpenStack `ansible-hardening` project, and runs it on `k8s-cluster` hosts when deploying.

See https://github.com/scality/metal-k8s/commit/130e337f922c429b20ef0aa56f6bf1ad7ace5daa for more information (other commits simply subtree the upstream repository).

### CODE REVIEW

File: metal-k8s.yml [hunk 1]
```diff
@@ -8,6 +8,11 @@
   roles:
     - role: python2_install
 
+- name: Harden all systems
+  hosts: k8s-cluster
+  roles:
+    - ansible-hardening
```
Comment (Zempashi) [inline metal-k8s.yml]: I would see something like:
`
roles:
    - role: ansible-hardening
      when: enable_hardening|default(True)|bool
`

To optionnaly opt-out hardening if needed (for developement purpose)
Note: for non-ansible-speaking person: It's enabled by default

**Comment by NicolasT** [comment]
@Zempashi What do you think?

File: ansible.cfg [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[defaults]
+roles_path = ansible-hardening
```
Comment (Zempashi) [inline ansible.cfg]: Why not puting ansible-hardening in the roles directory ?

**Comment by NicolasT** [comment]
Complete rework compared to the old version of this PR.

**Comment by NicolasT** [comment]
@ballot-scality @Zempashi One thing I noticed is this role enables SELinux, and Kubespray disables it again. I may look to not let this role enable SELinux in the first place.

**Pull request merged**